packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -251,17 +259,40 @@ class OpPerfTelemetry {
                 category,
                 pingLatency: this.pingLatency,
                 msnDistance: this.deltaManager.lastSequenceNumber - this.deltaManager.minimumSequenceNumber,
+                signalDuration,
```

Comment:
There is some value in sticking it all on same event, but I'd do it in the same way as we do with pingLatency - report last value, but single value, not averages.

That said, I'm not sure it would work out for signals. Pings happen all the time, so we know for sure we will have recent value to use. Signals are different - there will be payloads that have only signals (and no ops), and vice versa - ops and no signals. Recording only latest value will skew measurements to op-heavy workloads.

We somewhat have similar problem with pings, but that's the reason we report pings separately (see DeltaLatency event) and use that to do all kinds of statistical analyzes. The value on this even is only used to guess on how long it takes server to process ops (i.e. subtract network cost).

I'd say doing same with signals is needed. I.e. Signals report their own event (sampled) and report latest ping value as a properly on these events, to be able to calculate cost of processing signals in service.

Reply:
I will separate in the next iteration.